.ci/util.shlib

```diff
@@ -301,6 +301,30 @@ function UTIL_LOAD_CUSTOM_HOOK() {
         }
         popd > /dev/null || return 1
 	fi
+
+# $1: The repo name to parse
+# $2: The database URL
+function UTIL_PARSE_DATABASE() {
+    local repo="$1"
+    local db_url="$2"
+    local already_parsed=() # Prevent processing one pkgbase more than once
+
+    wget -q "${db_url}"
+    TEMP="$(mktemp -d)"
+    tar -xf "${repo}".db -C "$TEMP"
+
+    while read -r pkgbase version; do
+        # Escape any dots in the package name since that breaks the code
+        # because of an invalid arithmetic statement
+        normalized_pkgbase=${pkgbase//./}
+        if [[ "${already_parsed[*]}" == *"${normalized_pkgbase}"* ]]; then
+            continue
+        else
+            already_parsed+=("$normalized_pkgbase")
+            hash=$(echo -n "$version" | sha256sum | cut -d' ' -f1)
+            echo "$pkgbase:$hash" >>.ci/lib.state.new
```

Comment:
Copy pasted for visibility: You should write to TMPDIR instead of using .ci/ as TMPDIR. This way, it is completely impossible for temporary files to end up in the repository. Use mv to replace lib.state with lib.state.new once done.